File: tests/bindings/dotnet/foo.Tests/CollectionTest.cs
using System;
using Xunit;
using foo;
using System.Linq;
using System.Collections.Generic;

namespace foo.Tests
{
    public class CollectionTest
    {
        [Fact]
        public void StringCollectionTest()
        {
            var strings = new List<string>();
            strings.Add("Hello");
            strings.Add("World!");
            strings.Add("Émile");

            Assert.Equal((uint)strings.Count, StringCollectionMethods.GetSize(strings));
            Assert.Equal("Hello", StringCollectionMethods.GetValue(strings, 0));
            Assert.Equal("World!", StringCollectionMethods.GetValue(strings, 1));
            Assert.Equal("Émile", StringCollectionMethods.GetValue(strings, 2));
        }

        [Fact]
        public void StringCollectionWithReserveTest()
        {
            var strings = new List<string>();
            strings.Add("Hello");
            strings.Add("World!");
            strings.Add("Émile");

            Assert.Equal((uint)strings.Count, StringCollectionMethods.GetSizeWithReserve(strings));
            Assert.Equal("Hello", StringCollectionMethods.GetValueWithReserve(strings, 0));
            Assert.Equal("World!", StringCollectionMethods.GetValueWithReserve(strings, 1));
            Assert.Equal("Émile", StringCollectionMethods.GetValueWithReserve(strings, 2));
        }
    }
}

File: tests/bindings/java/foo-tests/src/test/java/io/stepfunc/foo_test/CollectionTest.java
package io.stepfunc.foo_test;

import io.stepfunc.foo.StringCollectionTestMethods;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;
import static org.joou.Unsigned.uint;

public class CollectionTest {
    @Test
    public void StringCollectionTest() {
        List<String> strings = new ArrayList<>();
        strings.add("Hello");
        strings.add("World!");
        strings.add("Émile");

        assertThat(StringCollectionTestMethods.getSize(strings)).isEqualTo(uint(3));
        assertThat(StringCollectionTestMethods.getValue(strings, uint(0))).isEqualTo("Hello");
        assertThat(StringCollectionTestMethods.getValue(strings, uint(1))).isEqualTo("World!");
        assertThat(StringCollectionTestMethods.getValue(strings, uint(2))).isEqualTo("Émile");
    }

    @Test
    public void StringCollectionWithReserveTest() {
        List<String> strings = new ArrayList<>();
        strings.add("Hello");
        strings.add("World!");
        strings.add("Émile");

        assertThat(StringCollectionTestMethods.getSizeWithReserve(strings)).isEqualTo(uint(3));
        assertThat(StringCollectionTestMethods.getValueWithReserve(strings, uint(0))).isEqualTo("Hello");
        assertThat(StringCollectionTestMethods.getValueWithReserve(strings, uint(1))).isEqualTo("World!");
        assertThat(StringCollectionTestMethods.getValueWithReserve(strings, uint(2))).isEqualTo("Émile");
    }
}
